lib/src/main/java/com/shopify/checkoutsheetkit/CheckoutWebViewEventProcessor.kt

```diff
@@ -40,8 +40,10 @@ internal class CheckoutWebViewEventProcessor(
     private val closeCheckoutDialogWithError: (CheckoutException) -> Unit = { CheckoutWebView.clearCache() },
     private val setProgressBarVisibility: (Int) -> Unit = {},
     private val updateProgressBarPercentage: (Int) -> Unit = {},
+    private val setCheckoutComplete: () -> Unit = {}
 ) {
     fun onCheckoutViewComplete(checkoutCompletedEvent: CheckoutCompletedEvent) {
+        setCheckoutComplete()
```

Comment:
Wouldn't it be better to call `CheckoutWebView.clearCache()` here to avoid keeping state in the CheckoutDialog?

Reply:
It has started to feel a little convoluted

The reason is, at the moment `CheckoutWebView.clearCache()` also destroys the WebView.  And we're not able to destroy the WebView when it's still attached to the view hierarchy. 

We could potentially stop destroying the WebView when clearing the cache, and call destroy in a few more places.

I can prototype it and see how it looks